docs/rest_api.md

```diff
@@ -416,6 +416,31 @@ Authorization requirements:
 Power cycle the node named `<node>`, and set it's next boot device to
 PXE. If the node is powered off, this turns it on.
 
+#### node_set_bootdev
+
+`POST /node/<node>/dev/<dev>/set_bootdev`
+
+Sets the nodes next boot device persistently
+
+device <dev> can be set to:
+
+  none  : Do not change boot device order
+  
+  pxe   : Force PXE boot
+  
+  disk  : Force boot from default Hard-drive
+  
+  safe  : Force boot from default Hard-drive, request Safe Mode
+  
+  diag  : Force boot from Diagnostic Partition
+  
+  cdrom : Force boot from CD/DVD
+  
+  bios  : Force boot into BIOS Setup
+  
+  floppy: Force boot from Floppy/primary removable media
```

Comment:
Questions:

1. Are all of these guaranteeded to be supported by an particular version of IPMI? we should be able to succicntly specify what requirements are for hardware to work with the IPMI driver.
2. Do we really want to allow access to all of these? I don't even know off-hand what exactly is meant by "diagnostic partition".
3. Should we actually enumerate the options in the spec? It would be nice for this interface to be portable to non-IPMI obms, and it would be nice not to have the "floppy" option be part of the stable API once we have one -- it's not clear it's relevant *now* and we want to be able to change what we support to match hardware. Maybe it would be better to have a way to fetch the available options, which may vary based on driver or hardware? @henn, @knikolla, @shwsun, curious to your thoughts on that point.